sycl/include/sycl/half_type.hpp

```diff
@@ -70,9 +70,9 @@ inline __SYCL_CONSTEXPR_HALF uint16_t float2Half(const float &Val) {
     Exp16 = 0x1f;
   } else if (__builtin_expect(Exp32Diff > -14, 0)) {
     // normal range for half type
-    Exp16 = Exp32Diff + 15;
+    Exp16 = static_cast<uint16_t>(Exp32Diff) + 15;
     // convert 23-bit mantissa to 10-bit mantissa.
-    Frac16 = Frac32 >> 13;
+    Frac16 = static_cast<uint16_t>(Frac32 >> 13);
```

Comment:
Note for reviewers: from what I understood (according to the cppreference), arguments of bitwise shift operator undergo integer promotion, i.e. if they are smaller than `int`, then they are implicitly casted to `int`. That effectively makes a result an `int` as well and triggers a warning about implicit narrowing conversion when we assign back to `int16_t`. Therefore, explicit casts were added to every place where we do bitwise shifts.

See https://cppreference.com/w/cpp/language/operator_arithmetic.html and https://cppreference.com/w/cpp/language/implicit_cast.html